addons/interaction/functions/fnc_passMagazine.sqf

```diff
@@ -0,0 +1,57 @@
+/*
+ * Author: BaerMitUmlaut
+ * Pass spare magazine for the specified weapon.
+ *
+ * Arguments:
+ * 0: Unit that passes the magazine <OBJECT>
+ * 1: Unit to pass the magazine to <OBJECT>
+ * 2: Weapon classname <STRING>
+ * 
+ * Return Value:
+ * None
+ *
+ * Example:
+ * [_player, _target, "arifle_MX_F"] call ace_interaction_fnc_magToPassazine
+ *
+ * Public: No
+ */
+
+#include "script_component.hpp"
+params ["_player", "_target", "_weapon"];
+private ["_compatibleMags", "_filteredMags", "_magToPass", "_magToPassIndex", "_playerName", "_magToPassDisplayName"];
+
+_compatibleMags = getArray (configfile >> "CfgWeapons" >> _weapon >> "magazines");
+_filteredMags = [magazinesAmmoFull _player, {
+    params ["_className", "", "_loaded"];
+    _className in _compatibleMags && !_loaded
+}] call EFUNC(common,filter);
+
+//select magazine with most ammo
+_magToPass = _filteredMags select 0;
+_magToPassIndex = 0;
+{
+    _x params ["_className", "_ammoCount"];
+    if ((_ammoCount > (_magToPass select 1)) && (_target canAdd _className)) then {
```

Comment:
Really minor thing, but if you `exitWith` here, you don't have to iterat through the rest of the array.